runtime/src/iree/hal/utils/stream_tracing.h

```diff
@@ -141,6 +141,15 @@ void iree_hal_stream_tracing_notify_submitted(
     iree_hal_stream_tracing_context_t* context,
     iree_hal_stream_tracing_context_event_list_t* event_list);
 
+// Alternatively, if you don't want the stream tracing to manually
+// handle collecting events, (because it may cause more blocking than
+// you would prefer), you can manually collect the events for
+// a specific event list so long as you never notified the submission.
+// You must still free the events to release them back into the context.
```

Comment:
```suggestion
// Manually collects the events for a specific event list.
// Callers must free the events to release them back into the context.
//
// Use this instead of `iree_hal_stream_tracing_notify_submitted` if you
// don't want the stream tracing to manually handle ... etc
```

Keep comments standalone so they are useful in isolation. People looking for the info on a single function shouldn't have to read the context around, if the ordering changes or the other methods are removed it still makes sense, and editors show just the comments on the method under inspection:
![image](https://github.com/user-attachments/assets/5a075cb4-22da-4f03-b940-22698604a614)

Function comments should start with a verb phrase: https://google.github.io/styleguide/cppguide.html#Function_Comments
(some good tips in there)